src/oop/special_class_method/hard/q6/MainTest.java

```diff
@@ -0,0 +1,264 @@
+                "The attribute " + stringFieldName + " could not be found in your class. Please make sure you have added it, it is spelled correctly, and has the correct type");
+        assertTrue(testClass.hasModifier(stringFieldName, "private"),
+                "Your " + stringFieldName + " does not have the correct visibility modifier.");
+    }
+
+    @Test
+    public void carClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {double.class, String.class};
+        assertTrue(testClass.hasConstructor(classArguments),
+                "Your " + className + " constructor does not have the correct parameters.");
+        assertTrue(testClass.hasModifier(classArguments, "public"),
+                "Your " + className + " constructor does not have the correct modifier.");
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, "Long"),
+                Arguments.of(234.4, "Round"),
+                Arguments.of(654.45, "Second"),
+                Arguments.of(43.3, "Fourth")
+
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void carClassHasCorrectConstructor(double value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, String.class}
+        };
+        Class<?>[] constructorClasses = {double.class, String.class};
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor does not initialize the " + doubleFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
+                "Your " + className + " constructor does not initialize the " + stringFieldName + " attribute to the correct value.";
+        String incorrectConstructorVisibilityModifier =
+                "Your " + className + " constructor does not have the correct visibility modifier.";
+        _assertEquals(value1, testClass.getFieldValue(classInstance, doubleFieldName),
+                incorrectDoubleValueMessage);
+        _assertEquals(value2, testClass.getFieldValue(classInstance, stringFieldName),
+                incorrectStringValueMessage);
+        assertTrue(testClass.hasModifier(constructorClasses, "public"), incorrectConstructorVisibilityModifier);
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void carClassHasCorrectConstructor2(double value1, String value2) throws Throwable {
+        Class<?>[] constructorClasses = {double.class, String.class};
+        Object classInstance = testClass.createInstance();
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor does not initialize the " + doubleFieldName + " attribute to the correct value.";
```

Comment:
I feel like since there are two constructors the error message should make it more clear which one is being referred to